testplugins/src/main/java/org/spongepowered/test/interact/ClickContainerEventCancelledTest.java

```diff
@@ -0,0 +1,78 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.test.interact;
+
+import com.google.inject.Inject;
+import net.kyori.adventure.identity.Identity;
+import net.kyori.adventure.text.Component;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.parameter.CommandContext;
+import org.spongepowered.api.event.EventContextKeys;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.item.inventory.container.ClickContainerEvent;
+import org.spongepowered.api.registry.RegistryTypes;
+import org.spongepowered.plugin.PluginContainer;
+import org.spongepowered.plugin.builtin.jvm.Plugin;
+import org.spongepowered.test.LoadableModule;
+
+/**
+ *   @author CanardNocturne
+ */
+@Plugin("clickcontainereventcancelledtest")
+public class ClickContainerEventCancelledTest implements LoadableModule {
+
+    private final PluginContainer plugin;
+
+    @Inject
+    public ClickContainerEventCancelledTest(final PluginContainer plugin) {
+        this.plugin = plugin;
+    }
+
+    @Override
+    public void enable(final CommandContext ctx) {
+        Sponge.eventManager().registerListeners(this.plugin, new ClickContainerEventCancelledTest.ClickContainerListener());
+    }
+
+    public static class ClickContainerListener {
+        @Listener
+        private void onInteractItem(final ClickContainerEvent.Primary event) {
+            event.setCancelled(true);
+            Sponge.server().broadcastAudience().sendMessage(Identity.nil(), Component.text("/*************"));
+            Sponge.server().broadcastAudience().sendMessage(Identity.nil(), Component.text().append(Component.text("/* Event: ")).append(Component.text(event.getClass().getSimpleName())).build());
+            Sponge.server().broadcastAudience().sendMessage(Identity.nil(),
+                    Component.text().append(Component.text("/* Hand: "))
+                            .append(Component.text(event.context().get(EventContextKeys.USED_HAND).map(h -> RegistryTypes.HAND_TYPE.keyFor(Sponge.game(), h).formatted()).orElse("UNKNOWN")))
+                            .build()
+            );
+            Sponge.game().systemSubject().sendMessage(Identity.nil(), Component.text().append(Component.text("/ Cause: ")).append(Component.text(event.cause().all().toString())).build());
```

Comment:
Out of curiosity, why the switch from `broadcastAudience` to `systemSubject`?

Reply:
This code is a copy of **InteractTest** with some modifications. I will change to `systemSubject` because `broadcastAudience` doesn't print anything.